content/learning-paths/servers-and-cloud-computing/exploiting-stack-buffer-overflow-aarch64/introduction.md

```diff
@@ -0,0 +1,54 @@
+---
+title: "Introduction: \"Smashing the stack\""
+weight: 2
+
+### FIXED, DO NOT MODIFY
+layout: learningpathall
+---
+
+In 1996, Phrack magazine published the article
+[Smashing the stack for fun and profit](http://phrack.org/issues/49/14.html#article).
+Since then, the article has become a milestone in memory vulnerability research
+and exploit development. It demonstrates how an attacker can exploit a buffer
+overflow on the stack to execute arbitrary code in a process that they only have
+access to by providing some inputs.
+
+It can be seen as the beginning of an arm's race between attackers and
+defenders. Each time, the defenders add more mitigations to try and block
+attackers. In turn, the attackers find new and more elaborate ways to
+work-around the hardening techniques the defenders implement.
+
+This arms race continues to this day. The importance of memory vulnerabilities
+and hardening against exploits making use of them has only grown. These
+presentations and blogs from
+[Microsoft](https://youtu.be/PjbGojjnBZQ?si=oCHCa0SHgaSNr6Gr&t=836),
+[Chromium](https://www.chromium.org/Home/chromium-security/memory-safety/) and
+[Android](https://security.googleblog.com/2021/01/data-driven-security-hardening-in.html)
+highlight that 60% to 70% of exploits they see are caused by a memory
+vulnerability.
+
+I found that to gain an in-depth understanding of how memory vulnerabilities are
+used to build an exploit; how the mitigations work; and how the arms race
+continues to drive new mitigations; it is very valuable to understand the state
+of the world at this "starting-point" in 1996.
+
+In this learning path, we will implement a few basic experiments to understand
+stack layout, buffer overflow and similar concepts, with the goal to build a
+simple end-to-end buffer overflow attack. We will do so on AArch64-linux.
+
+This learning path aims to give plenty of guidance for doing these
+implementations in as little time as possible, while not providing all answers,
```

Comment:
"not providing all answers" means not immediately providing all answers, right? I didn't see any questions without an answer provided (and having the answers is very useful at this introductory level).